google-cloud-clients/google-cloud-core/src/main/java/com/google/cloud/Date.java

```diff
@@ -57,13 +53,14 @@ public static Date fromYearMonthDay(int year, int month, int dayOfMonth) {
 
   /** @param date Data in RFC 3339 date format (yyyy-mm-dd). */
   public static Date parseDate(String date) {
-    Matcher matcher = FORMAT_REGEXP.matcher(date);
-    if (!matcher.matches()) {
-      throw new IllegalArgumentException("Invalid date: " + date);
-    }
-    int year = Integer.parseInt(matcher.group(1));
-    int month = Integer.parseInt(matcher.group(2));
-    int dayOfMonth = Integer.parseInt(matcher.group(3));
+    Preconditions.checkNotNull(date);
```

Comment:
This change seems straightforward.  Can you please break it out into a separate PR?

Reply:
Removed from this PR and submitted as separate PR.